tests/osaka/eip7934_block_rlp_limit/test_max_block_rlp_size.py

```diff
@@ -0,0 +1,341 @@
+        base_fee_per_gas="0x0",
+        withdrawals_root="0x" + "00" * 32,
+        blob_gas_used="0x0",
+        excess_blob_gas="0x0",
+        parent_beacon_block_root="0x" + "00" * 32,
+        requests_hash="0xe3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+    )
+
+
+def get_block_rlp_size(transactions: List[Transaction], gas_used: int) -> int:
+    """Calculate the RLP size of a block with given transactions."""
+    header = create_test_header(gas_used)
+    total_gas = sum((tx.gas_limit or 21000) for tx in transactions)
+    header.gas_used = ZeroPaddedHexNumber(total_gas)
+    test_block = FixtureBlockBase(blockHeader=header, withdrawals=[])  # type: ignore
+    return len(test_block.with_rlp(txs=transactions).rlp)
+
+
+@pytest.fixture
+def exact_size_transactions(
+    sender: EOA, block_size_limit: int, fork: Fork
+) -> Tuple[List[Transaction], int]:
+    """
+    Generate transactions that fill a block to exactly the RLP size limit.
+
+    The calculation uses caching to avoid recalculating the same block rlp for each
+    fork. Calculate the block and fill with real sender for testing.
+    """
+    stubbed_transactions, gas_used = _exact_size_transactions_calculation(block_size_limit, fork)
+    test_transactions = [
+        Transaction(
+            sender=sender,
+            nonce=tx.nonce,
+            max_fee_per_gas=tx.max_fee_per_gas,
+            max_priority_fee_per_gas=tx.max_priority_fee_per_gas,
+            gas_limit=tx.gas_limit,
+            data=tx.data,
+        )
+        for tx in stubbed_transactions
+    ]
+    return test_transactions, gas_used
+
+
+@lru_cache(maxsize=128)
+def _exact_size_transactions_calculation(
+    block_size_limit: int, fork: Fork
+) -> Tuple[List[Transaction], int]:
+    """Generate transactions that fill a block to exactly the RLP size limit."""
+    transactions = []
+    target_size = block_size_limit
+
+    sender = EOA("0x" + "00" * 20, key=123)  # stub account to fill the block
+    nonce = 0
+    total_gas_used = 0
+    max_block_gas = 100_000_000
+
+    calculator = fork.transaction_intrinsic_cost_calculator()
+
+    data_large = b"\x00" * 500_000
+    gas_limit_large = calculator(calldata=data_large)
+
+    # block with 16 transactions + large calldata remains safely below the limit
+    for _ in range(16):
+        tx = Transaction(
+            sender=sender,
+            nonce=nonce,
+            max_fee_per_gas=10**11,
+            max_priority_fee_per_gas=10**11,
+            gas_limit=gas_limit_large,
+            data=data_large,
+        )
+
+        transactions.append(tx)
+        total_gas_used += gas_limit_large
+        nonce += 1
+
+    current_size = get_block_rlp_size(transactions, gas_used=total_gas_used)
+    remaining_bytes = target_size - current_size
+    remaining_gas = max_block_gas - total_gas_used
+
+    if remaining_bytes > 0 and remaining_gas > 50_000:
+        # create and empty transaction to measure base contribution
```

Comment:
Note to self, fix typo.

```suggestion
        # create an empty transaction to measure base contribution
```